app/controllers/api/v1/articles_controller.rb

```diff
@@ -22,7 +22,7 @@ def create
   end
 
   def destroy
-    article = Article.find(params[:id])
+    article = Article.find_by(id: params[:id], user: current_user)
```

Comment:
`article` could be `nil` if there is no article that matches the search criteria. In this case, we should return a 404:

```suggestion
    article = Article.find_by(id: params[:id], user: current_user)
    
    unless article
      return render json: { error: 'article not found' }, status: :not_found
    end
```